### PR TITLE
feat(crc): Add support for CodeReady Containers

### PR DESCRIPTION
**Pull request by benoitf**
It fixes https://github.com/eclipse/che/issues/13940

image

Change-Id: If172e3b88df6f36aad09c1d69bab76ef5616fd2e
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

File: src/platforms/crc.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          const ip = await this.getCRCIP()
+          flags.domain = ip + '.nip.io'
+          task.title = `${task.title}...${flags.domain}.`
+        }
+      },
+    ], {renderer: flags['listr-renderer'] as any})
+  }
+
+  async isCRCRunning(): Promise<boolean> {
+    const { code, stdout } = await execa('crc', ['status'], { timeout: 60000, reject: false })
```
Comment (sunix) [inline src/platforms/crc.ts]: it looks like i don't have the status parameter with crc
```bash
$ crc version
version: 0.88.0-alpha-4.1.3+783056a

 $ crc status
Error: unknown command "status" for "crc"
Run 'crc --help' for usage.
FATA[0000] unknown command "status" for "crc"           


```

**Comment by sunix** [comment]
somehow had to create the `che` namespace manually
```
./bin/run server:start -p crc
  ✔ ✈️  CodeReady Containers preflight checklist
    ✔ Verify if oc is installed...done.
    ✔ Verify if crc is installed...done.
    ✔ Verify if CodeReady Containers is running...done.
    ✔ Retrieving CodeReady Containers IP and domain for routes URLs...192.168.130.11.nip.io.
  ❯ 🏃  Running the Che Operator
    ✔ Copying operator resources...done.
    ✔ Create Namespace (che)
    ✖ Create ServiceAccount che-operator in namespace che
      → namespaces "che" not found
```

File: src/platforms/crc.ts (hunk 1 above)
Comment (benoitf) [reply src/platforms/crc.ts]: I compiled CRC from master so probably you'll have it in the upcoming release

**Comment by rhopp** [comment]
This should wait after GA, right?

**Comment by sunix** [comment]
Yep I think it can wait. On another side it is a good way to test openshift 4 locally so good to have for developers as soon as we can.

**Comment by benoitf** [comment]
@rhopp could be a good way for QA to test on top of OpenShift 4 :-)

**Comment by rhopp** [comment]
@benoitf Correct me if I'm wrong, but to test Che/chectl on OCP 4 (or CRC), we don't need the CRC platform in chectl. We can use "openshift" platform, right? Only thing we will be missing is some pre-flight checks, right?

**Comment by benoitf** [comment]
@rhopp yes you can still use generic "openshift" platform but it's will ask you for a domain

**Comment by skabashnyuk** [comment]
@benoitf can you help me. I want to test this pr. But I have
```
[14:24:46]skabashn:bin[CHE-13940]#: ./run server:start --platform=crc --installer=operator
  ✔ ✈️  CodeReady Containers preflight checklist
    ✔ Verify if oc is installed...done.
    ✔ Verify if crc is installed...done.
    ✔ Verify if CodeReady Containers is running...done.
    ✔ Retrieving CodeReady Containers IP and domain for routes URLs...192.168.64.22.nip.io.
  ❯ 🏃‍  Running the Che Operator
    ✔ Copying operator resources...done.
    ✔ Create Namespace (che)
    ✖ Create ServiceAccount che-operator in namespace che
      → namespaces "che" not found
      Create Role che-operator in namespace che
      Create RoleBinding che-operator in namespace che
      Create CRD checlusters.org.eclipse.che
      Waiting 5 seconds for the new Kubernetes resources to get flushed
      Create deployment che-operator in namespace che
      Create Che Cluster eclipse-che in namespace che
Error: namespaces "che" not found
    at KubeHelper.<anonymous> (~/dev/src/redhat/chectl/src/api/kube.ts:126:13)
    at Generator.throw (<anonymous>)
    at rejected (~/dev/src/redhat/chectl/src/api/kube.ts:15:65)
```

**Comment by benoitf** [comment]
could you try by creating the namespace yes ?

If it works it means that the step 'create Che namespace' is not working

**Comment by skabashnyuk** [comment]
yes. it helps in some way. After the third attempt che was deployed.
```
[14:24:58]skabashn:bin[CHE-13940]#: oc new-project che
Now using project "che" on server "https://api.crc.testing:6443".

You can add applications to this project with the 'new-app' command. For example, try:

    oc new-app django-psql-example

to build a new example application in Python. Or use kubectl to deploy a simple Kubernetes application:

    kubectl create deployment hello-node --image=gcr.io/hello-minikube-zero-install/hello-node

[14:30:56]skabashn:bin[CHE-13940]#: ./run server:start --platform=crc --installer=operator
  ✔ ✈️  CodeReady Containers preflight checklist
    ✔ Verify if oc is installed...done.
    ✔ Verify if crc is installed...done.
    ✔ Verify if CodeReady Containers is running...done.
    ✔ Retrieving CodeReady Containers IP and domain for routes URLs...192.168.64.22.nip.io.
  ✔ 🏃‍  Running the Che Operator
    ✔ Copying operator resources...done.
    ✔ Create Namespace (che)...It already exist.
    ✔ Create ServiceAccount che-operator in namespace che...done.
    ✔ Create Role che-operator in namespace che...done.
    ✔ Create RoleBinding che-operator in namespace che...done.
    ✔ Create CRD checlusters.org.eclipse.che...done.
    ✔ Waiting 5 seconds for the new Kubernetes resources to get flushed...done.
    ✔ Create deployment che-operator in namespace che...done.
    ✔ Create Che Cluster eclipse-che in namespace che...done.
  ❯ ✅  Post installation checklist
    ✔ PostgreSQL pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ✔ Keycloak pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ❯ Che pod bootstrap
      ✔ scheduling...done.
      ✖ downloading images
        → Unauthorized
        starting
      Retrieving Che Server URL
      Che status check
Error: Unauthorized
    at KubeHelper.<anonymous> (~/dev/src/redhat/chectl/src/api/kube.ts:386:43)
    at Generator.throw (<anonymous>)
    at rejected (~/dev/src/redhat/chectl/src/api/kube.ts:15:65)
[14:40:18]skabashn:bin[CHE-13940]#: ./run server:start --platform=crc --installer=operator
  ✔ ✈️  CodeReady Containers preflight checklist
    ✔ Verify if oc is installed...done.
    ✔ Verify if crc is installed...done.
    ✔ Verify if CodeReady Containers is running...done.
    ✔ Retrieving CodeReady Containers IP and domain for routes URLs...192.168.64.22.nip.io.
  ✔ 🏃‍  Running the Che Operator
    ✔ Copying operator resources...done.
    ✔ Create Namespace (che)...It already exist.
    ✔ Create ServiceAccount che-operator in namespace che...It already exist.
    ✔ Create Role che-operator in namespace che...It already exist.
    ✔ Create RoleBinding che-operator in namespace che...It already exist.
    ✔ Create CRD checlusters.org.eclipse.che...It already exist.
    ✔ Waiting 5 seconds for the new Kubernetes resources to get flushed...done.
    ✔ Create deployment che-operator in namespace che...It already exist.
    ✔ Create Che Cluster eclipse-che in namespace che...It already exist.
  ✔ ✅  Post installation checklist
    ✔ PostgreSQL pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ✔ Keycloak pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ✔ Che pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ✔ Retrieving Che Server URL...http://che-che.apps-crc.testing
    ✔ Che status check
Command server:start has completed successfully.
```

**Comment by sunix** [comment]
che namespace creation issue maybe related to https://github.com/eclipse/che/issues/13981 ?

**Comment by skabashnyuk** [comment]
is it ready to be merged?

**Comment by skabashnyuk** [comment]
@benoitf can we merge this pr?

**Pull request merged**